swagger.yaml

```diff
@@ -0,0 +1,389 @@
+      - FlorenceAPIKey: []
+      responses:
+        200:
+          description: "The import job is in a queue"
+        400:
+          description: "Invalid json message was sent to the API"
+        404:
+          description: "InstanceId does not match any import jobs"
+        500:
+          description: "Failed to process the request due to an internal error"
+  /instance/{instanceId}:
+    get:
+      tags:
+      - "External Interface"
+      summary: "Get Information about an import job"
+      description: "Get the current state of an import job, this includes all events which have happened and a overall state of the job."
+      parameters:
+      - $ref: '#/parameters/instanceId'
+      produces:
+      - "application/json"
+      security:
+      - FlorenceAPIKey: []
+      responses:
+        200:
+          description: "Return a single import job"
+          schema:
+            $ref: '#/definitions/ImportJobState'
+        404:
+          description: "InstanceId does not match any import jobs"
+        500:
+          description: "Failed to process the request due to an internal error"
+  /instance/{instanceId}/events:
+    put:
+      tags:
+       - "Internal Interface"
+      summary: "Add an event to a import job"
+      description: |
+        Add a new event onto the import job. Events can be for information or error messages. Each event must contain a
+        type of event(Info or Error), a message to explaining the event, time of the event and finally the
+        message offset in kafka
+      parameters:
+      - $ref: '#/parameters/instanceId'
+      - $ref: '#/parameters/event'
+      security:
+      - InternalAPIKey: []
+      responses:
+        200:
+          description: "The event was added to the import job"
+        400:
+          description: "Invalid json message sent"
+        404:
+          description: "InstanceId does not match any import jobs"
+        500:
+          description: "Failed to process the request due to an internal error"
+  /instance/{instanceId}/dimensions:
+    put:
+      tags:
+      - "Internal Interface"
+      summary: "Add a dimension for the import job"
+      description: |
+        Add a dimension from the obervations found in the V3 data. To create a dimension the following items are needed;
+        * nodeName - The name of the node eg (instanceId)-(dimension).(value)
+        * nodeId - The unique id for this node
+      parameters:
+      - $ref: '#/parameters/instanceId'
+      - $ref: '#/parameters/dimension'
+      produces:
+      - "application/json"
+      security:
+      - InternalAPIKey: []
+      responses:
+        200:
+          description: "The dimension was added"
+        404:
+          description: "InstanceId does not match any import jobs"
+        500:
+          description: "Failed to process the request due to an internal error"
+    get:
+      tags:
+      - "Internal Interface"
+      summary: "Get all dimensions from an imported job"
```

Comment:
Again it seems import and instance are being interchanged even though they are two separate things.